src/config/config-set.ts

```diff
@@ -248,7 +246,9 @@ export class ConfigSet {
       this.logger.debug({ babelConfig: this.babelConfig }, 'normalized babel config via ts-jest option')
     }
     if (!this.babelConfig) {
-      this._overriddenCompilerOptions.module = this.compilerModule.ModuleKind.CommonJS
+      this._overriddenCompilerOptions.module = this.jestConfig.extensionsToTreatAsEsm.length
+        ? undefined
+        : this.compilerModule.ModuleKind.CommonJS
```

Comment:
I don't think `ts-jest` should care about this new option. Transformers will be called with `supportsStaticESM: true` if `import` statements can be used and `supportsDynamicImport: true` if `import` expressions can be used.

Expressions can be used in both ESM and CJS, but statements only work in ESM mode. I don't think you need to look at the stuff Jest uses to figure out what mode to run, you can just look at those flags

Reply:
aw thanks for the info. Actually I was looking for a single flag to decide to unforce.  With TypeScript compiler, it's not possible to transform codes with only `supportsStaticESM: true` or `supportsDynamicImport: true`. 

What do you suggest because it is possible that `supportsDynamicImport: false` and `supportsStaticESM: true` or vice versa. When will all ESM options be true ?

Reply:
hmm seem like I need to configure preset manually to enable all ESM options. I thought `extensionsToTreatAsEsm` can automatically enable all ESM options

Reply:
There is no single option to force either mode, as most likely CJS will always be a part of the test run (from `node_modules`). So it's always per file transformed. That said, it will be _all_ `.ts` files for instance

Reply:
I see, so you suggest that CJS should be compiled separately from non-CJS. Any extensions in `extensionsToTreatAsEsm` should be compiled with non-CJS path.

Reply:
yeah, sorta. I don't think you should look at `extensionsToTreatAsEsm` at all - you should look at the options that is passed to `process`

Reply:
Will ESM options be decided only via jest config or they can be also changed by runtime ?

Reply:
runtime, as we lookup `package.json` to check for https://nodejs.org/api/esm.html#esm_enabling